rectools/visuals/metrics_app.py

```diff
@@ -0,0 +1,120 @@
+import typing as tp
+
+import ipywidgets as widgets
+import pandas as pd
+import plotly.express as px
+import plotly.graph_objs as go
+from IPython.display import display
+
+from rectools import Columns
+
+DEFAULT_WIDTH = 800
+DEFAULT_HEIGHT = 600
+
+
+class MetricsApp:
+    """
+    Jupyter widgets app for metric visualization and comparison.
+    Do not create instances of this class directly. Use `MetricsApp.construct` method instead.
+    """
+
+    def __init__(
+        self,
+        df_metrics_data: pd.DataFrame,
+        show_legend: bool = True,
+        auto_display: bool = True,
+        plotly_kwargs: tp.Optional[tp.Dict[str, tp.Any]] = None,
+    ):
+        self.metrics_data = df_metrics_data
+        self.show_legend = show_legend
+        self.auto_display = auto_display
+        self.default_plotly_kwargs = {"width": DEFAULT_WIDTH, "height": DEFAULT_HEIGHT}
+        self.plotly_kwargs = {**self.default_plotly_kwargs, **(plotly_kwargs if plotly_kwargs is not None else {})}
+
+        self.fig = go.Figure()
```

Comment:
Do we need this only for linters or is it actually making something? I'm just curious, didn't work with plotly that much

Reply:
Only for mypy, it can't handle default None initialization

Reply:
You probably can do smth like [this](https://github.com/MobileTeleSystems/RecTools/blob/main/rectools/models/implicit_als.py#L64) - specify the type and don't specify the value

Reply:
Test it and found out that `self.fig = go.Figure()` is better option. It works as fallback if user create app with option `auto_display = False` and try to get figure `app.fig` without any widget interactions. If we don't specify `self.fig` value this would lead to error. I updated `happy_path`tests to make sure that we can handle `fig` extraction.